testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -5303,3 +5316,1287 @@ fn continue_tenure_extend() {
+        &sender_addr,
+        contract3_name,
+        "get-block-time",
+        vec![&clarity::vm::Value::UInt(last_stacks_block_height - 1)],
+    );
+    let time3a_block = time3a_block_value
+        .expect_optional()
+        .unwrap()
+        .unwrap()
+        .expect_u128()
+        .unwrap();
+    assert!(
+        time3a_block - time3_block >= 1,
+        "get-stacks-block-info? time should have changed"
+    );
+
+    // Sleep to ensure the seconds have changed
+    thread::sleep(Duration::from_secs(1));
+
+    // Mine a Nakamoto block
+    info!("Mining Nakamoto block");
+    let blocks_processed_before = coord_channel
+        .lock()
+        .expect("Mutex poisoned")
+        .get_stacks_blocks_processed();
+
+    // submit a tx so that the miner will mine an extra block
+    let transfer_tx =
+        make_stacks_transfer(&sender_sk, sender_nonce, send_fee, &recipient, send_amt);
+    submit_tx(&http_origin, &transfer_tx);
+
+    loop {
+        let blocks_processed = coord_channel
+            .lock()
+            .expect("Mutex poisoned")
+            .get_stacks_blocks_processed();
+        if blocks_processed > blocks_processed_before {
+            break;
+        }
+        thread::sleep(Duration::from_millis(100));
+    }
+
+    let time0b_value = call_read_only(
+        &naka_conf,
+        &sender_addr,
+        contract0_name,
+        "get-time",
+        vec![&clarity::vm::Value::UInt(last_stacks_block_height)],
+    );
+    let time0b = time0b_value
+        .expect_optional()
+        .unwrap()
+        .unwrap()
+        .expect_u128()
+        .unwrap();
+    assert_eq!(
+        time0a, time0b,
+        "get-block-info? time should not have changed"
+    );
+
+    let time1b_value = call_read_only(
+        &naka_conf,
+        &sender_addr,
+        contract1_name,
+        "get-time",
+        vec![&clarity::vm::Value::UInt(last_stacks_block_height)],
+    );
+    let time1b = time1b_value
+        .expect_optional()
+        .unwrap()
+        .unwrap()
+        .expect_u128()
+        .unwrap();
+    assert_eq!(
+        time0b, time1b,
+        "Time from pre- and post-epoch 3.0 contracts should match"
+    );
+
+    let time3b_block_value = call_read_only(
+        &naka_conf,
+        &sender_addr,
+        contract3_name,
+        "get-block-time",
+        vec![&clarity::vm::Value::UInt(last_stacks_block_height)],
+    );
+    let time3b_block = time3b_block_value
+        .expect_optional()
+        .unwrap()
+        .unwrap()
+        .expect_u128()
+        .unwrap();
+
+    assert!(
+        time3b_block - time3a_block >= 1,
+        "get-stacks-block-info? time should have changed"
+    );
+
+    coord_channel
+        .lock()
+        .expect("Mutex poisoned")
+        .stop_chains_coordinator();
+    run_loop_stopper.store(false, Ordering::SeqCst);
+
+    run_loop_thread.join().unwrap();
+}
+
+fn assert_block_info(
+    tuple0: &BTreeMap<ClarityName, Value>,
+    miner: &clarity::vm::Value,
```

Comment:
Why have the fully-qualified type name for `Value` here when you don't use it in the type parameter for `tuple0`?

Reply:
🤷 fixed in d2bc1276a.